### PR TITLE
added map<string, pair<double, map<int, double> > >

### PR DESCRIPTION
**Pull request by rwcarlsen**
Added new dbtype for upcoming separations facility.

### CODE REVIEW

**Comment by rwcarlsen** [comment]
This is build on #1108

File: src/sqlite_back.cc [hunk 1]
```diff
@@ -293,6 +293,7 @@ void SqliteBack::Bind(boost::spirit::hold_any v, DbTypes type, SqlStatement::Ptr
   CYCLUS_BINDVAL(MAP_STRING_STRING, std::map<std::string CYCLUS_COMMA std::string>);
   CYCLUS_BINDVAL(MAP_STRING_VECTOR_DOUBLE, std::map<std::string CYCLUS_COMMA std::vector<double> >);
   CYCLUS_BINDVAL(MAP_STRING_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::map<int CYCLUS_COMMA double> >);
+  CYCLUS_BINDVAL(MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::pair<double CYCLUS_COMMA std::map<int CYCLUS_COMMA double> > >);
```
Comment (gidden) [inline src/sqlite_back.cc]: `len(line) > 80`, same for others.

**Comment by gidden** [comment]
Can you add a test, please?

File: src/sqlite_back.cc (hunk 1 above)
Comment (rwcarlsen) [inline src/sqlite_back.cc]: Here is the diff:

```diff
   CYCLUS_BINDVAL(MAP_INT_STRING, std::map<int CYCLUS_COMMA std::string>);
   CYCLUS_BINDVAL(MAP_STRING_INT, std::map<std::string CYCLUS_COMMA int>);
   CYCLUS_BINDVAL(MAP_STRING_DOUBLE, std::map<std::string CYCLUS_COMMA double>);
-  CYCLUS_BINDVAL(MAP_STRING_STRING, std::map<std::string CYCLUS_COMMA std::string>);
-  CYCLUS_BINDVAL(MAP_STRING_VECTOR_DOUBLE, std::map<std::string CYCLUS_COMMA std::vector<double> >);
-  CYCLUS_BINDVAL(MAP_STRING_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::map<int CYCLUS_COMMA double> >);
-  CYCLUS_BINDVAL(MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::pair<double CYCLUS_COMMA std::map<int CYCLUS_COMMA double> > >);
+  CYCLUS_BINDVAL(MAP_STRING_STRING,
+                 std::map<std::string CYCLUS_COMMA std::string>);
+  CYCLUS_BINDVAL(MAP_STRING_VECTOR_DOUBLE,
+                 std::map<std::string CYCLUS_COMMA std::vector<double> >);
+  CYCLUS_BINDVAL(
+      MAP_STRING_MAP_INT_DOUBLE,
+      std::map<std::string CYCLUS_COMMA std::map<int CYCLUS_COMMA double> >);
+  CYCLUS_BINDVAL(MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE,
+                 std::map<std::string CYCLUS_COMMA std::pair<
+                     double CYCLUS_COMMA std::map<int CYCLUS_COMMA double> > >)
```

I'm willing to make the change if you think the code is clearer/better with the line breaks.  I think it looks/functions better as is FWIW.

Comment (rwcarlsen) [inline src/sqlite_back.cc]: The diff was produced using the output of clang-format with the google style guide configuration.

Comment (rwcarlsen) [inline src/sqlite_back.cc]: I'll just do it - not worth the discussion time :-)

Comment (gidden) [inline src/sqlite_back.cc]: Thanks @rwcarlsen. To be frank, I think they both look bad; in such cases,
I default to the style guide.

On Mon, Mar 30, 2015 at 5:16 PM, Robert Carlsen <notifications@github.com>
wrote:

> In src/sqlite_back.cc
> <https://github.com/cyclus/cyclus/pull/1109#discussion_r27437272>:
>
> > @@ -293,6 +293,7 @@ void SqliteBack::Bind(boost::spirit::hold_any v, DbTypes type, SqlStatement::Ptr
> >    CYCLUS_BINDVAL(MAP_STRING_STRING, std::map<std::string CYCLUS_COMMA std::string>);
> >    CYCLUS_BINDVAL(MAP_STRING_VECTOR_DOUBLE, std::map<std::string CYCLUS_COMMA std::vector<double> >);
> >    CYCLUS_BINDVAL(MAP_STRING_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::map<int CYCLUS_COMMA double> >);
> > +  CYCLUS_BINDVAL(MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::pair<double CYCLUS_COMMA std::map<int CYCLUS_COMMA double> > >);
>
> I'll just do it - not worth the discussion time :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1109/files#r27437272>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by rwcarlsen** [comment]
test added.

File: src/sqlite_back.cc [hunk 2]
```diff
@@ -364,6 +371,7 @@ boost::spirit::hold_any SqliteBack::ColAsVal(SqlStatement::Ptr stmt,
   CYCLUS_LOADVAL(MAP_STRING_STRING, std::map<std::string CYCLUS_COMMA std::string>);
   CYCLUS_LOADVAL(MAP_STRING_VECTOR_DOUBLE, std::map<std::string CYCLUS_COMMA std::vector<double> >);
   CYCLUS_LOADVAL(MAP_STRING_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::map<int CYCLUS_COMMA double> >);
+  CYCLUS_LOADVAL(MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE, std::map<std::string CYCLUS_COMMA std::pair<double CYCLUS_COMMA std::map<int CYCLUS_COMMA double> > >);
```
Comment (gidden) [inline src/sqlite_back.cc]: line length as well

File: src/sqlite_back.cc [hunk 3]
```diff
@@ -445,6 +453,7 @@ DbTypes SqliteBack::Type(boost::spirit::hold_any v) {
     type_map[&typeid(std::map<std::string, std::string>)] = MAP_STRING_STRING;
     type_map[&typeid(std::map<std::string, std::vector<double> >)] = MAP_STRING_VECTOR_DOUBLE;
     type_map[&typeid(std::map<std::string, std::map<int, double> >)] = MAP_STRING_MAP_INT_DOUBLE;
+    type_map[&typeid(std::map<std::string, std::pair<double, std::map<int, double> > >)] = MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE;
```
Comment (gidden) [inline src/sqlite_back.cc]: line length as well

**Comment by gidden** [comment]
this passes tests for me locally

File: src/sqlite_back.cc (hunk 3 above)
Comment (rwcarlsen) [inline src/sqlite_back.cc]: done.

**Comment by gidden** [comment]
this needs a rebase now

**Comment by rwcarlsen** [comment]
rebased.

File: src/sqlite_back.cc (hunk 2 above)
Comment (rwcarlsen) [inline src/sqlite_back.cc]: done

**Comment by gidden** [comment]
looks good and passes locally, waiting on CI

**Comment by gidden** [comment]
merged

**Pull request merged**